lib/room.cpp

```diff
@@ -1293,6 +1313,17 @@ const StateEventBase* Room::getCurrentState(const QString& evtType,
     return d->getCurrentState({ evtType, stateKey });
 }
 
+const QList<const StateEventBase*>*
```

Comment:
That's not quite what I was looking for: returning a dynamically allocated `QList` will surely lead to people forgetting to `delete` it afterwards, let alone being inefficient since `QList` is [implicitly-shared](https://doc.qt.io/qt-5/implicit-sharing.html) anyway. Also - `QVector` is equally or more efficient compared to `QList` and [should be used by default](https://doc.qt.io/qt-5/qlist.html#details) (in fact, Qt 6 has `QVector` and `QList` referring to the same class, and it's basically Qt 5's `QVector`).
```suggestion
QVector<const StateEventBase*>
```